Build failure when using boost::asio

I am using Ubuntu 24.04 with the distro's boost dev package.

When building glaze, some of the tests fail to compile (http_chunked_test, http_client_ssl_test, http_client_no_ssl_executor_test).

Part of the problem is that these tests use the `asio` namespace but the tests themselves are not in the `glz` namespace.  As a result, they do not see `glz::asio` nor `boost::asio`.

Doing a search-and-replace to change `asio::` to `glz::asio::` results in more errors.   Here are a few:
```
/home/cout/git/glaze/tests/networking_tests/http_chunked_test/http_chunked_test.cpp:87:29: error: no match for ‘operator!=’ (operand types are ‘std::error_code’ and ‘boost::asio::error::misc_errors’)
   87 |          if (running_ && ec != glz::asio::error::eof && ec != glz::asio::error::operation_aborted) {
      |                          ~~ ^~ ~~~~~~~~~~~~~~~~~~~~~
      |                          |                       |
      |                          std::error_code         boost::asio::error::misc_errors
/home/cout/git/glaze/tests/networking_tests/http_chunked_test/http_chunked_test.cpp:87:60: error: no match for ‘operator!=’ (operand types are ‘std::error_code’ and ‘boost::asio::error::basic_errors’)
   87 |          if (running_ && ec != glz::asio::error::eof && ec != glz::asio::error::operation_aborted) {
      |                                                         ~~ ^~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                         |                       |
      |                                                         std::error_code         boost::asio::error::basic_errors
/home/cout/git/glaze/tests/networking_tests/http_client_ssl_test/http_client_ssl_test.cpp:248:11: error: no match for ‘operator==’ (operand types are ‘const std::error_code’ and ‘boost::asio::error::misc_errors’)
  248 |    if (ec == asio::error::eof) return true;
      |        ~~ ^~ ~~~~~~~~~~~~~~~~
      |        |                  |
      |        |                  boost::asio::error::misc_errors
      |        const std::error_code
/home/cout/git/glaze/tests/networking_tests/http_client_ssl_test/http_client_ssl_test.cpp:249:11: error: no match for ‘operator==’ (operand types are ‘const std::error_code’ and ‘boost::asio::error::basic_errors’)
  249 |    if (ec == asio::error::connection_reset) return true;
      |        ~~ ^~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |        |                  |
      |        |                  boost::asio::error::basic_errors
      |        const std::error_code
```

I am unsure how to resolve this -- are these comparisons valid in a newer version of boost?  Ubuntu 24.04 ships with 1.83.

--- a/tests/networking_tests/http_chunked_test/http_chunked_test.cpp
+++ b/tests/networking_tests/http_chunked_test/http_chunked_test.cpp
@@ -13,6 +13,14 @@
 #include "glaze/net/http_server.hpp"
 #include "ut/ut.hpp"
 
+#if defined(GLZ_USING_BOOST_ASIO)
+namespace asio
+{
+   using namespace boost::asio;
+   using error_code = boost::system::error_code;
+}
+#endif
+
 using namespace ut;
 
 // --------------------------------------------------------------------------
@@ -84,7 +92,8 @@ struct chunked_test_server
    void setup_routes()
    {
       server_.on_error([this](std::error_code ec, std::source_location) {
-         if (running_ && ec != asio::error::eof && ec != asio::error::operation_aborted) {
+         if (running_ && ec != make_error_code(asio::error::eof) &&
+             ec != make_error_code(asio::error::operation_aborted)) {
             std::fprintf(stderr, "Server error: %s\n", ec.message().c_str());
          }
       });

--- a/tests/networking_tests/http_client_no_ssl_executor_test/http_client_no_ssl_executor_test.cpp
+++ b/tests/networking_tests/http_client_no_ssl_executor_test/http_client_no_ssl_executor_test.cpp
@@ -12,6 +12,14 @@
 #include "glaze/net/http_client.hpp"
 #include "ut/ut.hpp"
 
+#if defined(GLZ_USING_BOOST_ASIO)
+namespace asio
+{
+   using namespace boost::asio;
+   using error_code = boost::system::error_code;
+}
+#endif
+
 using namespace ut;
 
 suite http_client_no_ssl_executor_tests = [] {

--- a/tests/networking_tests/http_client_ssl_test/http_client_ssl_test.cpp
+++ b/tests/networking_tests/http_client_ssl_test/http_client_ssl_test.cpp
@@ -245,11 +245,11 @@ struct TestResponse
 // Helper to suppress expected errors
 bool should_suppress_error(const std::error_code& ec)
 {
-   if (ec == asio::error::eof) return true;
-   if (ec == asio::error::connection_reset) return true;
-   if (ec == asio::error::operation_aborted) return true;
-   if (ec == asio::error::bad_descriptor) return true;
-   if (ec == asio::error::broken_pipe) return true;
+   if (ec == make_error_code(asio::error::eof)) return true;
+   if (ec == make_error_code(asio::error::connection_reset)) return true;
+   if (ec == make_error_code(asio::error::operation_aborted)) return true;
+   if (ec == make_error_code(asio::error::bad_descriptor)) return true;
+   if (ec == make_error_code(asio::error::broken_pipe)) return true;
    return false;
 }
 

